blog/_posts/2025-07-02-kaito-distributed-inference.md

```diff
@@ -0,0 +1,92 @@
+- Have access to commercial GPUs like NVIDIA H100 or A100
+- Want to avoid per-token API fees (which easily skyrocket costs at scale)
+- Keen to fine-tune or customize the model — something closed APIs usually block
+- Have sensitive or proprietary data to keep ringfenced and protected from accidental exposure through third-party logs
+
+Self-hosting with the Kubernetes AI Toolchain Operator (KAITO) helps you achieve all this and more! KAITO is a CNCF Sandbox project that simplifies and optimizes your inference and tuning workloads on Kubernetes. It integrates with vLLM, a high-throughput LLM inference engine optimized for serving large models efficiently. 
+
+vLLM supports quantized models (like 4-bit GPTQ or AWQ), reducing memory/GPU requirements drastically without major accuracy trade-offs. KAITO’s modular, plug-and-play setup allows you to go from model selection to production-grade API quickly:
+
+- Out-of-the-box OpenAI-compatible API means you can swap in KAITO with minimal application-side changes.
+- Built-in support for prompt formatting, batching, and streaming responses.
+- Self-hosting with KAITO on your AKS cluster ensures data never leaves your organization's controlled environment, ideal for highly regulated industries (finance, healthcare, defense) where cloud LLM APIs may be restricted due to compliance.
+
+# The catch? Managing huge model weights 🏋️‍♂️
+Some models come with **massive** weight files, and even when fully quantized can weigh hundreds of gigabytes (based on model type and version). Handling and deploying such model serving workloads isn’t trivial, especially if you want reproducible, scalable workflows on Kubernetes.
+
+KAITO balances simplicity and efficiency but using container images to manage most LLMs - but it can become difficult to distribute large model files in heavy-weight images.
+
+Luckily, KAITO inferencing now supports these model weights with the power of a local file cache and striped Non-Volatile Memory Express (NVMe) PersistentVolume managed by Azure Container Storage v2.
+
+> A local file cache significantly reduces latency during model downloads and reads, enhances reliability with persistent storage and avoids repetitive downloads after container restarts - all without extra storage fees! 
+
+# What is Azure Container Storage?
+
+[Azure Container Storage](https://learn.microsoft.com/en-us/azure/storage/container-storage/container-storage-introduction) (ACStor) is a cloud-based volume management, deployment, and orchestration service built natively for containers. It integrates with Kubernetes, allowing you to dynamically and automatically provision persistent volumes to store data for stateful applications running on Kubernetes clusters.
+
+ACStor v2 is purpose-built for AI and high-performance computing (HPC) workloads that demand ultra-fast data processing on local NVMe disks. It delivers performance close to raw NVMe speeds, all while providing seamless Kubernetes-native operations.

```

Comment:
We introduced it in this section without `v2` so I think the version is unecessary here.